data/generalhw_scripts/gadget-control.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/python3
+#
+# Documentation: https://www.kernel.org/doc/Documentation/usb/gadget_hid.txt
+# Based on https://gist.github.com/marmarek/5c44ffeb2f36e106b4d34e7e8780c208
```

Comment:
This got several fixes/improvements since previous posting. I've just updated the gist.

Reply:
You couldn't share [this line](https://github.com/os-autoinst/os-autoinst/pull/1898/files#diff-457f23a16e93b308d1b29031db3d25ab7f7b5bc10b63541230aa51891523b9cdR88) then: 
But I'm fine with duplicating `mouse_button` function for that. @Martchus any preference? You originally requested to de-duplicate this part.

Reply:
Context: VNC and USB HID have different bits order there, see https://gist.github.com/marmarek/5c44ffeb2f36e106b4d34e7e8780c208/revisions#diff-f92d3a59d8f9df9a7d6415db914f4403016a1bd6388444b25ba57db889633444R345-R353

Reply:
Does it mean that VNC uses the current ordering and this is not an internal arbitrary convention for `os-autoinst`?

Reply:
Yes.

Reply:
`mouse_button` looks already de-duplicated on https://github.com/os-autoinst/os-autoinst/pull/1898. Am I missing something? In general I'd say de-duplicating is always a good idea but keeping backwards compatibility is more important. Your code doesn't look like it breaks backwards compatibility, though.

Reply:
It's about undoing this deduplication, to use different bits order in the button bitmask. Currently I need to [re-arrange them when sending to USB HID](https://gist.github.com/marmarek/5c44ffeb2f36e106b4d34e7e8780c208/revisions#diff-f92d3a59d8f9df9a7d6415db914f4403016a1bd6388444b25ba57db889633444R345-R353).

Reply:
I'm fine with it either way. If you like to undo the de-duplication to avoid the re-arrangement then go for it.

Reply:
Workaround removed from this script, since it will be fixed/handled in `os-autoinst` directly.